ihmc-high-level-behaviors/src/main/java/us/ihmc/behaviors/sequence/actions/FootstepPlanActionPlanningThread.java

```diff
@@ -0,0 +1,161 @@
+package us.ihmc.behaviors.sequence.actions;
+
+import us.ihmc.avatar.drcRobot.ROS2SyncedRobotModel;
+import us.ihmc.commons.FormattingTools;
+import us.ihmc.commons.thread.TypedNotification;
+import us.ihmc.euclid.referenceFrame.FramePose3D;
+import us.ihmc.footstepPlanning.FootstepPlan;
+import us.ihmc.footstepPlanning.FootstepPlannerOutput;
+import us.ihmc.footstepPlanning.FootstepPlannerRequest;
+import us.ihmc.footstepPlanning.FootstepPlanningModule;
+import us.ihmc.footstepPlanning.graphSearch.graph.visualization.BipedalFootstepPlannerNodeRejectionReason;
+import us.ihmc.footstepPlanning.graphSearch.parameters.InitialStanceSide;
+import us.ihmc.footstepPlanning.log.FootstepPlannerLogger;
+import us.ihmc.footstepPlanning.tools.FootstepPlannerRejectionReasonReport;
+import us.ihmc.robotics.robotSide.RobotSide;
+import us.ihmc.robotics.robotSide.SideDependentList;
+
+public class FootstepPlanActionPlanningThread
```

Comment:
I dont see a thread, why is it labeled as a thread?

Reply:
Nevermind, I see you're creating a thread each update loop. Can you change it to where there is only one thread, as the class name implies, and have the `triggerPlan` method submit new jobs to run?

Reply:
Yeah so preview plans are triggered once a second while the action is next for execution. Sometimes a prior plan can run long and a newer one can be done faster, so we want to allow to plan in parallel to improve the user experience.

Also the plan for execution is done always once when the action execution is triggered. We want this to be very independent of the preview planning.